test/integration/gardenlet/vpaevictionrequirements/vpaevictionrequirements_test.go

```diff
@@ -0,0 +1,217 @@
+						})),
+					}))
+				}).Should(Succeed())
+			})
+		})
+		When("the Shoot is inside its maintenance window", func() {
+			BeforeEach(func() {
+				metav1.SetMetaDataAnnotation(&vpa.ObjectMeta, v1beta1constants.AnnotationShootMaintenanceWindow, maintenanceWindowNow.Begin+","+maintenanceWindowNow.End)
+				vpa.Spec.UpdatePolicy.EvictionRequirements = upscaleOnlyRequirement
+			})
+			It("should remove the EvictionRequirement and requeue for the end of the maintenance window", func() {
+				Eventually(func(g Gomega) {
+					reconciledVPA := &vpaautoscalingv1.VerticalPodAutoscaler{}
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), reconciledVPA)).To(Succeed())
+					g.Expect(reconciledVPA.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": BeEmpty(),
+						})),
+					}))
+				}).Should(Succeed())
+			})
+		})
+		When("the Shoot maintenance window is updated", func() {
+			BeforeEach(func() {
+				metav1.SetMetaDataAnnotation(&vpa.ObjectMeta, v1beta1constants.AnnotationShootMaintenanceWindow, maintenanceWindowNow.Begin+","+maintenanceWindowNow.End)
+				vpa.Spec.UpdatePolicy.EvictionRequirements = upscaleOnlyRequirement
+			})
+			It("reconciles the VPA again", func() {
+				Eventually(func(g Gomega) {
+					reconciledVPA := &vpaautoscalingv1.VerticalPodAutoscaler{}
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), reconciledVPA)).To(Succeed())
+					g.Expect(reconciledVPA.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": BeEmpty(),
+						})),
+					}))
+				}).Should(Succeed())
+
+				Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), vpa)).To(Succeed())
+				metav1.SetMetaDataAnnotation(&vpa.ObjectMeta, v1beta1constants.AnnotationShootMaintenanceWindow, maintenanceWindowAlreadyPassed.Begin+","+maintenanceWindowAlreadyPassed.End)
+				Expect(testClient.Update(ctx, vpa)).To(Succeed())
+
+				Eventually(func(g Gomega) {
+					reconciledVPA := &vpaautoscalingv1.VerticalPodAutoscaler{}
+					g.Expect(testClient.Get(ctx, client.ObjectKeyFromObject(vpa), reconciledVPA)).To(Succeed())
+					g.Expect(reconciledVPA.Spec).To(MatchFields(IgnoreExtras, Fields{
+						"UpdatePolicy": PointTo(MatchFields(IgnoreExtras, Fields{
+							"EvictionRequirements": ConsistOf(upscaleOnlyRequirement),
+						})),
+					}))
+				}).Should(Succeed())
+
+			})
+		})
+		//When("the VPA is deleted", func() {
+		//	It("doesn't reconcile the VPA", func() {
+		//
+		//	})
+		//})
+	})
+
+	Context("VPA is annotated with downscale-never", func() {
+		BeforeEach(func() {
+			metav1.SetMetaDataLabel(&vpa.ObjectMeta, v1beta1constants.LabelVPAEvictionRequirementDownscaleRestriction, v1beta1constants.EvictionRequirementNever)
+		})
+		It("should add an EvictionRequirement and not requeue, regardless of a Shoot's maintenance window", func() {
```

Comment:
Maybe add another case for what happens when the maintenance time window is now?

Reply:
done